torch/onnx/utils.py

```diff
@@ -1787,64 +1758,58 @@ def _run_symbolic_function(
     namespace, op_name = ns_op_name.split("::")
 
     try:
-        symbolic_registry.register_version("", opset_version)
-
         # Caffe2-specific: Quantized op symbolics are registered for opset 9 only.
         if symbolic_helper.is_caffe2_aten_fallback() and opset_version == 9:
             symbolic_caffe2.register_quantized_ops("caffe2", opset_version)
 
-        if namespace == "aten":
-            domain = ""
-        elif namespace == "quantized" and symbolic_helper.is_caffe2_aten_fallback():
+        if namespace == "quantized" and symbolic_helper.is_caffe2_aten_fallback():
             domain = "caffe2"
         else:
             domain = namespace
+        symbolic_function_name = f"{domain}::{op_name}"
 
-        if symbolic_registry.is_registered_op(op_name, domain, opset_version):
-            symbolic_fn = _find_symbolic_in_registry(
-                domain, op_name, opset_version, operator_export_type
-            )
-            assert symbolic_fn is not None
-
-            attrs = {k: symbolic_helper._node_get(n, k) for k in n.attributeNames()}
-            if _need_symbolic_context(symbolic_fn):
-                ctx = _exporter_states.SymbolicContext(_params_dict, env, n, block)
-                return symbolic_fn(ctx, g, *inputs, **attrs)
-            # PythonOp symbolic need access to the node to resolve the name conflict,
-            # this is inconsistent with regular op symbolic.
-            if op_name == "PythonOp":
-                inputs = (n, *inputs)
-            return symbolic_fn(g, *inputs, **attrs)
-        elif namespace == "onnx":
+        symbolic_function_group = registration.registry.get_function_group(
+            symbolic_function_name
+        )
+        if symbolic_function_group is not None:
+            symbolic_fn = symbolic_function_group.get(opset_version)
```

Comment:
Per offline discussion, we will circle back in the future to refine this extra layer of `symbolic_function_group`. The main selling point is for handling more complicated scenarios, like function overloads, other than just opset version dispatching.